src/index.ts

```diff
@@ -123,9 +123,10 @@ export const htmlPlugin = (configuration: Configuration = { files: [], }): esbui
         const templateContext = {
             define: htmlFileConfiguration.define,
         }
-
-        const compiledTemplateFn = lodashTemplate(htmlTemplate)
-        return compiledTemplateFn(templateContext)
+        const compiledTemplateFn = htmlTemplate.toString().endsWith('.html') 
+            ? lodashTemplate(fs.readFileSync(htmlTemplate).toString()) 
```

Comment:
Let us also check if the file exists using `fs.exists`.

I also think some people are still calling their html files just `.htm`.